esbulk.go

```diff
@@ -0,0 +1,200 @@
+package esbulk
+
+import (
+	"bufio"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+)
+
+// Options represents bulk indexing options.
+type Options struct {
+	Servers     []string
+	Index       string
+	Purge       bool
+	Mapping     string
+	DocType     string
+	NumWorkers  int
+	ZeroReplica bool
+	GZipped     bool
+	BatchSize   int
+	Verbose     bool
+	IDField     string
+	Scheme      string // http or https; deprecated, use: Servers.
+	Username    string
+	Password    string
+	MaxRetries  int
+}
+
+// ProcessLDJFile reads input file and creates an index given options using
+// multiple workers
+func ProcessLDJFile(r io.Reader, options Options) (int, error) {
```

Comment:
How about `CreateIndexFromLDJ`? Seems more descriptive than `Process*`

Reply:
Only downside that I see with using `Create*` is that it also deletes index (if specified) but agree that `Process*` is too generic name.